### PR TITLE
Update loopback-mode type

### PR DESCRIPTION
**Pull request by sallylsy**
### Change Scope

* In openconfig-transport-types.yang, updates the current enums of loopback-mode-type.
* Two of the existing enums are updated to new values. If users were using FACILITY or TERMINAL, they should update to ASIC_MAC_LOCAL/PHY_MAC_LOCAL or ASIC_MAC_REMOTE/PHY_MAC_REMOTE

Ref: 

[Reference to SAI implementation](https://github.com/opencomputeproject/SAI/blob/a9070a5b597c6636772080e811d65b6e47ddda76/inc/saiport.h#L133)

[SAI header](https://github.com/opencomputeproject/SAI/pull/1934)

[Arista EOS - traffic loopback](https://www.arista.com/en/um-eos/eos-data-transfer#xx1332706) supports local and remote options.  It also allows specifying MAC and PHY layer options.

[JunOS - set loopback](https://www.juniper.net/documentation/us/en/software/junos/interfaces-ethernet/topics/topic-map/ethernet-10-40-100-gigabit-loopback-testing.html#id-configure-a-local-loopback) supports local loopback only.  JunOS does not expose a MAC or PHY level option from their configuration

[Cisco IOS XR - loopback](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/Interfaces/b-interfaces-hardware-component-cr-8000/ethernet-interface-commands.html#wp2680511456) supports local (aka terminal or internal) and remote (aka facility or line) loopbacks.   Cisco does not expose a MAC or PHY level option from their configuration.

### CODE REVIEW

File: release/models/optical-transport/openconfig-transport-types.yang [hunk 1]
```diff
@@ -179,17 +185,29 @@ module openconfig-transport-types {
         description
           "No loopback is applied";
       }
-      enum FACILITY {
```
[the remark this answers is not in the document]
Comment (sallylsy) [reply release/models/optical-transport/openconfig-transport-types.yang:191]: Updated.

**Comment by dplore** [comment]
We will review in OC operators meeting on Jan 23, 2024

**Comment by dplore** [comment]
I am not a deep expert on transceiver loopbacks, but I thought some transceiver modules implement a PHY level loopback in the transceiver module (as opposed to the ASIC).  Is this true?

@jnelson-sj do you have experience here?  Does it make sense to have some loopback enum for a transceiver PHY level loopback?

**Comment by jnelson-sj** [comment]
+Aditya Bhiday ***@***.***>
Hi Darren,

Yes, there are transceivers which perform loopbacks at different levels of
the processing chain.  For example, ZR transceivers have the following
loopbacks defined in OIF-400GZR:

[image: image.png]
You can (hopefully) see in the diagram that there are 6 loopbacks defined
here.  The can be grouped into whether they provide a FACILITY or a
TERMINAL loopback:

FACILITY Loopbacks:
- Media Side Tx
- Tx Modem
- Host Side Tx

TERMINAL Loopbacks:
- Media Side Rx
- Rx Modem
- Host Side Rx

I think the PHY level loopback you are talking about may be the Host Side
Tx - what the switch port sends on its tx is looped back to its rx with
minimal processing in the transceiver.

I wonder if there should be another dimension to the loopbacks - a depth.
The FACILITY/TERMINAL gives the direction of the loopback, but it would
also be good to have a DEEP/SHALLOW qualifier.  This would be used to
specify how much of the transceiver function is to be exercised in the
loopback.

Is this along the lines of what you are thinking?

John

**Comment by dplore** [comment]
Hi John, thank you for the insight!  My bias is to handle the 400ZR case separately as I think the ASIC MAC/PHY layer loopbacks proposed here are still applicable and do appear to conflict with the 400ZR case.

**Comment by dplore** [comment]
This was reviewed in the Jan 30, 2024 OC Operators meeting.  The feedback was we should address how a system which does not expose MAC / PHY layer loopbacks, but only an 'interface' loopback should be handled.   The suggestion was to 
 retain the existing `FACILITY` and `TERMINAL` enum values rather than deprecating them.

**Review by dplore — APPROVED** [review]
LGTM.  Last call for comments please.  This will merge on Feb 13, 2024

**Pull request merged**